example/examplechecker.py

```diff
@@ -1,128 +1,190 @@
 #!/usr/bin/env python3
 # from src.enochecker import *
-from enochecker import *
+import json
+import secrets
+from typing import Dict
+
+from enochecker import BaseChecker, BrokenServiceException, assert_equals, run
 
 
 class ExampleChecker(BaseChecker):
     """
     Change the methods given here, then simply create the class and .run() it.
-    Magic.
 
     A few convenient methods and helpers are provided in the BaseChecker.
-    ensure_bytes ans ensure_unicode to make sure strings are always equal.
-
-    As well as methods:
-    self.connect() connects to the remote server.
-    self.get and self.post request from http.
-    self.team_db is a dict that stores its contents to filesystem. (call .persist() to make sure)
-    self.readline_expect(): fails if it's not read correctly
+    When using an HTTP client (requests) or a plain TCP connection (telnetlib) use the
+    built-in functions of the BaseChecker that include some basic error-handling.
 
-    To read the whole docu and find more goodies, run python -m pydoc enochecker
-    (Or read the source, Luke)
+    The full documentation is available at https://enowars.github.io/enochecker/
     """
 
-    flag_count = 1
+    # how many flags does this service deploy per round? each flag should be stored at a different location in the service
+    flag_count = 2
+    # how many noises does this service deploy per round?
     noise_count = 1
+    # how many different havoc methods does this service use per round?
     havoc_count = 1
 
-    port = (
-        80
-    )  # The port will automatically be picked up as default by self.connect and self.http.
+    # The port will automatically be picked up as default by self.connect and self.http methods.
+    port = 80
+
+    def login(self, credentials) -> None:
+        res = self.http_post("/login", json=credentials)
+        assert_equals(res.status_code, 200)
+
+    def register_and_login(self, credentials) -> None:
+        res = self.http_post("/register", json=credentials)
+        assert_equals(res.status_code, 200)
 
-    def putflag(self):  # type: () -> None
+        self.login(credentials)
+
+    def generate_credentials(self) -> Dict[str, str]:
+        credentials = {
+            "username": secrets.token_urlsafe(12),
+            "password": secrets.token_urlsafe(16),
+        }
+        self.team_db[self.flag] = credentials
```

Comment:
Just a minor nitpick: I would make this method side-effect free and explicitly set the creds in putflag. 
Pros: 
- the method can be tested independently
- It's clear to the reader of putflag (ctf novice) that we have to write to the db.

Cons:
- one more line of code after each `generate_credentials`

Reply:
makes sense

Reply:
resolved in #53